contivk8s/k8s_cni.go

```diff
@@ -0,0 +1,129 @@
+/***
+Copyright 2016 Cisco Systems Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/contiv/netplugin/contivk8s/clients"
+	"github.com/contiv/netplugin/mgmtfn/cniapi"
+
+	logger "github.com/Sirupsen/logrus"
+)
+
+var log *logger.Entry
+
+func getPodInfo(ppInfo *cniapi.CNIPodAttr) error {
+	cniArgs := os.Getenv("CNI_ARGS")
+	if cniArgs == "" {
+		return fmt.Errorf("Error reading CNI_ARGS")
+	}
+
+	// convert the cniArgs to json format
+	cniArgs = "{\"" + cniArgs + "\"}"
+	cniTmp1 := strings.Replace(cniArgs, "=", "\":\"", -1)
+	cniJSON := strings.Replace(cniTmp1, ";", "\",\"", -1)
+	err := json.Unmarshal([]byte(cniJSON), ppInfo)
+	if err != nil {
+		return fmt.Errorf("Error parsing cni args: %s", err)
+	}
+
+	// nwNameSpace and ifname are passed as separate env vars
+	ppInfo.NwNameSpace = os.Getenv("CNI_NETNS")
+	ppInfo.IntfName = os.Getenv("CNI_IFNAME")
+	return nil
+}
+
+func addPodToContiv(nc *clients.NWClient, pInfo *cniapi.CNIPodAttr) {
+
+	// Add to contiv network
+	result, err := nc.AddPod(pInfo)
+	if err != nil {
+		log.Fatalf("EP create failed -- %s", err)
```

Comment:
kubelet just checks for status 0 and treats everything else as an error without interpreting it.